site/src/Util/NotificationHydrator.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Util;
+
+/**
+ * This is the data class for creating a notification
+ * which will be then sent out to the RabbitMQ queue
+ *
+ * @copyright Aquib Baig <aquibbaig97@gmail.com>
+ *
+ * Class NotificationHydrator
+ */
+class NotificationHydrator
+{
+    /**
+     * @var string $subject
+     */
+    protected $subject;
+
+    /**
+     * @var string $link
+     */
+    protected $link;
+
+    /**
+     * @var string $message
+     */
+    protected $message;
+
+    /**
+     * @var string $type
+     */
+    protected $type;
+
+    /**
+     * @var int $userIdentifier
+     */
+    protected $userIdentifier;
```

Comment:
Is this the sender or the recipient? And it should be of type `User`; you can serialize that into a user id when sending.